[py-sentry-sdk] added version 1.5.5

qwertos opened the pull request.

qwertos committed:
[py-sentry-sdk] audited dependencies and relisted extras as variants

qwertos committed:
[py-sentry-sdk] added version 1.5.5

qwertos committed:
[py-sentry-sdk] flake8

adamjstewart commented on var/spack/repos/builtin/packages/py-sentry-sdk/package.py:
```diff
@@ -12,23 +12,51 @@ class PySentrySdk(PythonPackage):
-    depends_on('py-pyspark@2.4.4:', type=('build', 'run'))
-    depends_on('py-pure-eval', type=('build', 'run'))
-    depends_on('py-chalice@1.16.0:', type=('build', 'run'))
+    variant('flask', default=False)
+    variant('quart', default=False)
+    variant('bottle', default=False)
+    variant('falcon', default=False)
+    variant('django', default=False)
+    variant('sanic', default=False)
+    variant('celery', default=False)
+    variant('beam', default=False)
+    variant('rq', default=False)
+    variant('aiohttp', default=False)
+    variant('tornado', default=False)
+    variant('sqlalchemy', default=False)
+    variant('pyspark', default=False)
+    variant('pure_eval', default=False)
+    variant('chalice', default=False)
+    variant('httpx', default=False)
```
These variants need a description

qwertos committed:
[py-sentry-sdk]

- added "descriptions" to variants
- removed conflicts
  - replaced with when statements in varients

alalazo commented:
```
1. py-sentry-sdk: dependency on py-httpx@0.16.0: cannot be satisfied by known versions of py-httpx
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-sentry-sdk/package.py
    known versions of py-httpx are 0.15.2, 0.11.1
```
